From e8cc7eb758100ed5f772868781e31ea5bef17f6b Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Wed, 5 Sep 2018 16:20:10 +0300
Subject: [PATCH 4586/5242] MLK-19460-2: drm: imx: dcss: remove PM_QoS

commit  8266848f578efdda38768a0aa54be86c867d010c from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch removes PM_QoS request from DCSS driver. This will allow the
A-53 cores to go idle even when DCSS is used.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-crtc.c |    4 ----
 drivers/gpu/imx/dcss/dcss-common.c   |   17 -----------------
 drivers/gpu/imx/dcss/dcss-prv.h      |    1 -
 include/video/imx-dcss.h             |    3 ---
 4 files changed, 25 deletions(-)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-crtc.c b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
index 832fa2d..fd169e9 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-crtc.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
@@ -109,8 +109,6 @@ static int dcss_enable_vblank(struct drm_crtc *crtc)
 
 	dcss_crtc->irq_enabled = true;
 
-	dcss_req_pm_qos(dcss, true);
-
 	dcss_vblank_irq_enable(dcss, true);
 
 	enable_irq(dcss_crtc->irq);
@@ -128,8 +126,6 @@ static void dcss_disable_vblank(struct drm_crtc *crtc)
 
 	dcss_vblank_irq_enable(dcss, false);
 
-	dcss_req_pm_qos(dcss, false);
-
 	dcss_crtc->irq_enabled = false;
 }
 
diff --git a/drivers/gpu/imx/dcss/dcss-common.c b/drivers/gpu/imx/dcss/dcss-common.c
index 2c05d9a..d8094a1 100644
--- a/drivers/gpu/imx/dcss/dcss-common.c
+++ b/drivers/gpu/imx/dcss/dcss-common.c
@@ -618,23 +618,6 @@ static int dcss_remove(struct platform_device *pdev)
 	return 0;
 }
 
-void dcss_req_pm_qos(struct dcss_soc *dcss, bool en)
-{
-	if (en && !dcss->pm_req_active) {
-		pm_qos_add_request(&dcss->pm_qos_req,
-				   PM_QOS_CPU_DMA_LATENCY, 0);
-		dcss->pm_req_active = true;
-		return;
-	}
-
-	if (dcss_dtrc_is_running(dcss, 1) || dcss_dtrc_is_running(dcss, 2))
-		return;
-
-	pm_qos_remove_request(&dcss->pm_qos_req);
-	dcss->pm_req_active = false;
-}
-EXPORT_SYMBOL(dcss_req_pm_qos);
-
 #ifdef CONFIG_PM_SLEEP
 static int dcss_suspend(struct device *dev)
 {
diff --git a/drivers/gpu/imx/dcss/dcss-prv.h b/drivers/gpu/imx/dcss/dcss-prv.h
index 4e5ddc2..e8c6f08 100644
--- a/drivers/gpu/imx/dcss/dcss-prv.h
+++ b/drivers/gpu/imx/dcss/dcss-prv.h
@@ -61,7 +61,6 @@ struct dcss_soc {
 	bool clks_on;
 
 	struct pm_qos_request pm_qos_req;
-	bool pm_req_active;
 };
 
 /* BLKCTL */
diff --git a/include/video/imx-dcss.h b/include/video/imx-dcss.h
index 51a3615..232f125 100644
--- a/include/video/imx-dcss.h
+++ b/include/video/imx-dcss.h
@@ -53,9 +53,6 @@ struct dcss_client_platformdata {
 
 #define dcss_trace_module(mod_tag, val) dcss_trace_write((mod_tag) | (val));
 
-/* COMMON */
-void dcss_req_pm_qos(struct dcss_soc *dcss, bool en);
-
 /* BLKCTL */
 void dcss_blkctl_hdmi_secure_src_en(struct dcss_soc *dcss);
 
-- 
1.7.9.5

